src/meta/src/manager/cluster.rs

```diff
@@ -174,6 +202,45 @@ where
         Ok(())
     }
 
+    pub async fn update_schedulability(
+        &self,
+        host_address: HostAddress,
+        is_unschedulable: bool,
+    ) -> MetaResult<WorkerType> {
+        let mut core = self.core.write().await;
+        let worker = core
+            .workers
+            .get_mut(&WorkerKey(host_address.clone()))
+            .ok_or_else(|| anyhow!("Worker node does not exist!"))?;
+        let worker_type = worker.worker_type();
+
+        // TODO
+        // We need to handle the deleting state once we introduce it
+        // if worker_type == WorkerType::ComputeNode && worker.worker_node.state == State::DELETING
```

Comment:
Deleted the comment. 

FYI @yezizp2012: `DELETING` state may be useful for the deletion workflow. Currently delete works like this: You send a delete gRPC to meta, saying that you want to delete node `X`. Meta "forgets" about `X`. `X` sends a heartbeat. Meta responds with an error and on receiving this response `X` crashes. **For cloud this is a problem**, because that means that the pod on which the CN resides will restart. The restarted pod will host a new CN, which will register itself with Meta. This is why we can never really scale down using the delete gRPC 😓 

We would prefer an orderly approach to that. The approach that would be nice for cloud looks as follows: 
- The delete workflow will cordon `X` and no more new actors are placed on `X`
- The delete workflow sends a delete request to Meta
- `X` is marked as `DELETING`
- `X` still sends heartbeats to meta
- On receiving heartbeats from `X`, meta will respond normally and `X` will not crash. The pod on which `X` is will not restart
- The delete workflow rebalances the actors. `X` is now marked as unschedulable and has no actors 
- The delete workflow removes the pod of `X`
- Meta missed 3 heartbeats of `X` and considers the `DELETING` node as deleted and forgets about it

CC @ALeitert @arkbriar

Reply:
It's quite tricky to introduce something like 3 time heartbeat here, the whole process would be much more simpler as follows:
1. the delete workflow will cordon `X` and no more new actors are placed on `X`.
2. rebalances all actors from `X` to others.
3. shutdown pod `X`, and then send `delete` request to Meta.

Reply:
That may be a viable approach as well. IMHO wether or not this works will depend on if Meta can treat a deleting node just like a regular node. 

FYI: I played around with the "missing 3 heartbeats and then considering the CN down" logic [here](https://github.com/risingwavelabs/risingwave/pull/9253/files#r1235178366).

Reply:
> It's quite tricky to introduce something like 3 times heartbeat here, the whole process would be much more simpler as follows:
> 
> 1. the delete workflow will cordon `X` and no more new actors are placed on `X`.
> 2. rebalances all actors from `X` to others.
> 3. shutdown pod `X`, and then send `delete` request to Meta.

I agree with @yezizp2012 that this way might be more clear. A residual `DELETING` state may affect the next scaling out unexpectedly.